File: aspnet-in-practice/04/VB/MasterPages-Advanced/App_Code/BasePage.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.IO

Public Class BasePage
	Inherits Page
	Protected Overloads Overrides Sub OnPreInit(ByVal e As EventArgs)
		Dim masterPage As String = File.ReadAllText(Server.MapPath("~/App_Data/MasterPage.config"))
		If Not String.IsNullOrEmpty(masterPage) Then
			MasterPageFile = masterPage
		End If

		MyBase.OnPreInit(e)
	End Sub
End Class

File: aspnet-in-practice/04/CS/MasterPages-Advanced/App_Code/BasePage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.IO;

public class BasePage: Page
{
    protected override void OnPreInit(EventArgs e)
    {
        string masterPage = File.ReadAllText(Server.MapPath("~/App_Data/MasterPage.config"));
        if (!string.IsNullOrEmpty(masterPage))
        {
            MasterPageFile = masterPage;
        }

        base.OnPreInit(e);
    }
}